### PR TITLE
exception_handling / it-academy.by / JC1-117-22 / Aliaksei Kharko / Done tasks 01-07

### PR DESCRIPTION
**Pull request by KharkoAO**
## Pull request state

Actual status for PR:
- [x] implemented
- [x] reviewed
- [ ] fixed
- [ ] approved

### CODE REVIEW

File: src/test/java/com/rakovets/course/java/core/practice/exception_handling/JavaUtilLogging/JavaUtilLogging.java [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package com.rakovets.course.java.core.practice.exception_handling.JavaUtilLogging;
+
+import java.util.logging.Logger;
+
+public class JavaUtilLogging {
```
Comment (rakovets) [inline src/test/java/com/rakovets/course/java/core/practice/exception_handling/JavaUtilLogging/JavaUtilLogging.java:5]: remove it

File: src/test/java/com/rakovets/course/java/core/practice/exception_handling/Task01Demo.java [hunk 2]
```diff
@@ -0,0 +1,15 @@
+package com.rakovets.course.java.core.practice.exception_handling;
+
+import static com.rakovets.course.java.core.practice.exception_handling.JavaUtilLogging.JavaUtilLogging.logger;
+
+public class Task01Demo {
```
Comment (rakovets) [inline src/test/java/com/rakovets/course/java/core/practice/exception_handling/Task01Demo.java:5]: add logger for each class

File: src/main/java/com/rakovets/course/java/core/practice/exception_handling/Task04.java [hunk 3]
```diff
@@ -0,0 +1,13 @@
+package com.rakovets.course.java.core.practice.exception_handling;
+
+import com.rakovets.course.java.core.practice.exception_handling.exception.ZeroCheckException;
+
+public class Task04 {
+    public int calculatorDividing(int dividend, int divisor) throws ZeroCheckException {
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/exception_handling/Task04.java:6]:
```suggestion
    public int calculatorDividing(int dividend, int divisor) {
```

File: src/test/java/com/rakovets/course/java/core/practice/exception_handling/Task04Demo.java [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+public class Task04Demo {
+    public static void main(String[] args) {
+        Task04 task = new Task04();
+        int dividend = 8;
+        int divisor = 0;
+
+        try {
+            logger.info(String.valueOf(task.calculatorDividing(dividend, divisor)));
+        } catch (ZeroCheckException ex) {
+            ex.printStackTrace();
```
Comment (rakovets) [inline src/test/java/com/rakovets/course/java/core/practice/exception_handling/Task04Demo.java:16]: use logger for stacktrace

File: src/test/java/com/rakovets/course/java/core/practice/exception_handling/Task03Demo.java [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+public class Task03Demo {
+    public static void main(String[] args) {
+        Task03 task = new Task03();
+        String userName = "Us";
+
+        try {
+            task.longCheckUserName(userName);
+            logger.info(userName);
+        } catch (LongCheckUserNameException ex) {
+            logger.severe(ex.getMessage());
```
Comment (rakovets) [inline src/test/java/com/rakovets/course/java/core/practice/exception_handling/Task03Demo.java:16]: where is stacktrace?

File: src/test/java/com/rakovets/course/java/core/practice/exception_handling/Task06Demo.java [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+        String[] array = {"2", "0", "6", "a", null};
+        String processingNumber = array[1];
+
+        try {
+            task.checkCorrectEnteredNumber(processingNumber);
+        } catch (ArithmeticException | NumberFormatException ex) {
+            logger.severe(ex.getMessage());
+        } catch (NullPointerException ex) {
+            logger.severe(ex.getMessage());
+        }
```
Comment (rakovets) [inline src/test/java/com/rakovets/course/java/core/practice/exception_handling/Task06Demo.java:18]: wrong solution

**Review by rakovets — CHANGES_REQUESTED** [review]
You need fix and add correct solutions

File: src/test/java/com/rakovets/course/java/core/practice/exception_handling/JavaUtilLogging/JavaUtilLogging.java (hunk 1 above)
Comment (KharkoAO) [reply src/test/java/com/rakovets/course/java/core/practice/exception_handling/JavaUtilLogging/JavaUtilLogging.java]: this fixed

File: src/test/java/com/rakovets/course/java/core/practice/exception_handling/Task01Demo.java (hunk 2 above)
Comment (KharkoAO) [reply src/test/java/com/rakovets/course/java/core/practice/exception_handling/Task01Demo.java:5]: this fixed

File: src/test/java/com/rakovets/course/java/core/practice/exception_handling/Task04Demo.java (hunk 4 above)
Comment (KharkoAO) [reply src/test/java/com/rakovets/course/java/core/practice/exception_handling/Task04Demo.java]: this fixed

File: src/test/java/com/rakovets/course/java/core/practice/exception_handling/Task03Demo.java (hunk 5 above)
Comment (KharkoAO) [reply src/test/java/com/rakovets/course/java/core/practice/exception_handling/Task03Demo.java:19]: this fixed

File: src/test/java/com/rakovets/course/java/core/practice/exception_handling/Task06Demo.java (hunk 6 above)
Comment (KharkoAO) [reply src/test/java/com/rakovets/course/java/core/practice/exception_handling/Task06Demo.java]: this fixed

**Comment by rakovets** [comment]
outdated

**Pull request closed without merging**